packages/experimental/Button/src/Button.tsx

```diff
@@ -7,14 +7,21 @@ import { stylingSettings, getDefaultSize } from './Button.styling';
 import { compose, mergeProps, withSlots, UseSlots } from '@fluentui-react-native/framework';
 import { useButton } from './useButton';
 import { Icon } from '@fluentui-react-native/icon';
-import { createIconProps } from '@fluentui-react-native/interactive-hooks';
+import { createIconProps, IPressableState } from '@fluentui-react-native/interactive-hooks';
 
-export const buttonLookup = (layer: string, state: ButtonState, userProps: ButtonProps): boolean => {
+/**
+ * A function which determines if a set of styles should be applied to the compoent given the current state and props of the button.
+ *
+ * @param layer The name of the state that is being checked for
+ * @param state The current state of the button
+ * @param userProps The props that were passed into the button
+ * @returns Whether the styles that are assigned to the layer should be applied to the button
+ */
+export const buttonLookup = (layer: string, state: IPressableState, userProps: ButtonProps): boolean => {
   return (
     state[layer] ||
     userProps[layer] ||
-    layer === userProps['size'] ||
-    (!userProps.fab && layer === getDefaultSize()) ||
+    (!userProps.fab && (layer === userProps['size'] || (!userProps['size'] && layer === getDefaultSize()))) ||
```

Comment:
Why the special casing around FAB? FWIW, mobile does have multiple size fab buttons, so if the goal was to only have one size fab button I would not do that.

Reply:
Mostly lack of redlines/focusing more on web/win32 redlines. I didn't want to accidentally mess with FAB in a weird way either, so in order to scope the change I excluded it.

Reply:
The FAB as is written isn't exactly to redlines anyway, since things like size tokens didn't exist when we made it. So I wouldn't worry about it _too_ much but fair!
Figma for reference: https://www.figma.com/file/ArWQiU7BAlrDGNmbrGEXbA/[Specs]-Fluent-vNext-iOS-Components?node-id=2024%3A497